source/jormungandr/jormungandr/interfaces/v1/serializer/places.py

```diff
@@ -0,0 +1,287 @@
+        lon = str(coordinates[0])
+        lat = str(coordinates[1])
+    else:
+        lon = None
+        lat = None
+    return lon, lat
+
+
+class CoordField(serpy.Field):
+    def as_getter(self, serializer_field_name, serializer_cls):
+        return lambda obj: self.generate_coord_field(obj)
+
+    def generate_coord_field(self, obj):
+        lon, lat = get_lon_lat(obj)
+        return {
+            "lat": lat,
+            "lon": lon,
+        }
+
+
+class CoordId(serpy.Field):
+    def as_getter(self, serializer_field_name, serializer_cls):
+        return lambda obj: self.generate_coord_id(obj)
+
+    def generate_coord_id(self, obj):
+        if not obj:
+            return None
+        lon, lat = get_lon_lat(obj)
+        return '{};{}'.format(lon, lat)
+
+
+class IntPropertyPathField(serpy.Field):
+    def as_getter(self, serializer_field_name, serializer_cls):
+        return lambda v: self.get_property(v, self.attr)
+
+    def get_property(self, obj, path):
+        properties = path.split(("."))
+        for property_ in properties:
+            result = obj.get(property_, {})
+            obj = result
+
+        return obj
+
+    def to_value(self, value):
+        return int(value) if value else None
+
+
+class SubAdministrativeRegionField(serpy.DictSerializer):
+    id = serpy.Field()
+    insee = serpy.Field()
+    name = serpy.Field(attr="label")
+    label = serpy.Field()
+    level = serpy.MethodField()
+    coord = serpy.MethodField()
+    zip_code = serpy.MethodField()
+
+    def get_coord(self, obj):
+        coord = obj.get('coord', {})
+        lat = str(coord.get('lat')) if coord and coord.get('lat') else None
+        lon = str(coord.get('lon')) if coord and coord.get('lon') else None
+        return {
+            "lat": lat,
+            "lon": lon
+        }
+
+    def get_level(self, obj):
+        return int(obj.get('level')) if obj.get('level') else None
+
+    def get_zip_code(self, obj):
+        zip_codes = obj.get('zip_codes', [])
+        if all(zip_code == "" for zip_code in zip_codes):
+            return None
+        elif len(zip_codes) == 1:
+            return zip_codes[0]
+        else:
+            return '{}-{}'.format(min(zip_codes), max(zip_codes))
+
+
+class AdministrativeRegionSerializer(serpy.DictSerializer):
+    id = PropertyPathField(attr='properties.geocoding.id')
+    name = PropertyPathField(attr='properties.geocoding.name')
+    label = PropertyPathField(attr='properties.geocoding.label')
+    zip_code = PropertyPathField(attr='properties.geocoding.postcode')
+    coord = CoordField()
+    insee = serpy.MethodField()
+    level = IntPropertyPathField(attr='properties.geocoding.level')
+    administrative_regions = serpy.MethodField()
+
+    def get_insee(self, obj):
+        geocoding = obj.get('properties', {}).get('geocoding', {})
+        return geocoding.get('citycode') or geocoding.get('city_code')
+
+    def get_administrative_regions(self, obj):
```

Comment:
SubAdministrativeRegionField is used in AddressSerializer and StopAreaSerializer